src/iterative_sorting/iterative_sorting.py

```diff
@@ -1,27 +1,44 @@
 # TO-DO: Complete the selection_sort() function below 
+arr = [643,44,32,1,25,56,73,8,34,69,23,67]
+
 def selection_sort( arr ):
     # loop through n-1 elements
     for i in range(0, len(arr) - 1):
-        cur_index = i
-        smallest_index = cur_index
+        current_index = i
+        # smallest_index = current_index
         # TO-DO: find next smallest element
         # (hint, can do in 3 loc) 
-             
-
-
-
+        for j in range(i+1, len(arr)):
+            if arr[j] < arr[current_index]:
+                current_index = j
+            
         # TO-DO: swap
-
-
-
+        temp = arr[i]
+        arr[i] = arr[current_index]
+        arr[current_index] = temp
 
     return arr
 
+# print(selection_sort(arr))
+    # temp = i
+    # i = j
+    # j = temp
 
 # TO-DO:  implement the Bubble Sort function below
 def bubble_sort( arr ):
+    for i in range(len(arr) - 1):
+        for j in range(len(arr) - 1):  
+            if arr[j] > arr[j+1]:
+                temp = arr[j]
+                arr[j] = arr[j+1]
+                arr[j+1] = temp
 
     return arr
+# arr = [643,44,32,1,25,56,73,8,34,69,23,67]
```

Comment:
In this circumstance you are never using i, but you are looping through enough times that you will end up sorting the input array/list.

However, your function will run all the way through both loops even if the array/list is sorted well before that.
Another approach to consider for this algorithm is using a boolean to keep track of whether or not you made a swap in your inner loop.
```
while boolean:
    boolean = false;
    do one bubble pass through full array
          if you make a swap:  boolean = true
          // if you didnt make a swap, your array is sorted and you exit the while loop without any more unnecessary iterations
```